src/hawtio/plugins/shared/mbean/node.ts

```diff
@@ -1,73 +1,12 @@
 import { Logger } from '@hawtio/core'
+import React from 'react'
+import { CubeIcon, FolderIcon, FolderOpenIcon } from '@patternfly/react-icons'
+import { TreeViewDataItem } from '@patternfly/react-core'
 import { escapeDots, escapeTags } from '@hawtio/util/jolokia'
+import { IJmxMBean } from 'jolokia.js'
 import { stringSorter, trimQuotes } from '@hawtio/util/strings'
-import { TreeViewDataItem } from '@patternfly/react-core'
-import { CubeIcon, FolderIcon, FolderOpenIcon } from '@patternfly/react-icons'
-import { IJmxDomain, IJmxDomains, IJmxMBean } from 'jolokia.js'
-import React from 'react'
-import { pluginName } from '../globals'
-
-const log = Logger.get(`${pluginName}-tree`)
-
-export class MBeanTree {
-  private tree: MBeanNode[] = []
-
-  constructor(domains: IJmxDomains) {
-    this.populate(domains)
-  }
-
-  private populate(domains: IJmxDomains) {
-    Object.entries(domains).forEach(([name, domain]) => {
-      // Domain name is displayed in the tree, so let's escape it here.
-      // Use a custom escaping method here as escaping '"' breaks Camel tree.
-      const escapedName = escapeTags(name)
-      this.populateDomain(escapedName, domain)
-    })
-
-    this.sortTree()
-
-    // TODO: tree post processors
-  }
-
-  private populateDomain(name: string, domain: IJmxDomain) {
-    log.debug("JMX tree domain:", name)
-    const domainNode = this.getOrCreateNode(name)
-    Object.entries(domain).forEach(([propList, mbean]) => {
-      domainNode.populateMBean(propList, mbean)
-    })
-  }
-
-  private getOrCreateNode(name: string): MBeanNode {
-    const node = this.tree.find(node => node.name === name)
-    if (node) {
-      return node
-    }
-
-    const id = escapeDots(name)
-    const newNode = new MBeanNode(id, name, true)
-    this.tree.push(newNode)
-    return newNode
-  }
-
-  private sortTree() {
-    this.tree.sort((a, b) => stringSorter(a.name, b.name))
-    this.tree.forEach(node => node.sort(true))
-  }
-
-  getTree(): MBeanNode[] {
-    return this.tree
-  }
 
-  isEmpty(): boolean {
-    return this.tree.length === 0
-  }
-}
-
-const Icons = {
-  folder: React.createElement(FolderIcon),
-  folderOpen: React.createElement(FolderOpenIcon),
-  mbean: React.createElement(CubeIcon),
-} as const
+const log = Logger.get(`mbean-node`)
```

Comment:
We use the naming convention `hawtio-<plugin>-<area>` for loggers. Let's follow the convention.

Reply:
Yeah. Was not happy with this after posting the PR yesterday. Since added an id to the tree so that the logger gets the pluginName where the tree is used rather than just "shared". That was we can separate, for example, 'jmx-tree-logger' from 'camel-tree-logger'.